semgrep-core/src/runner/Run_semgrep.ml

```diff
@@ -452,11 +452,16 @@ let targets_of_config (config : Runner_config.t)
  * It takes a set of rules and a set of targets and
  * recursively process those targets.
  *)
-let semgrep_with_rules config ((rules, skipped_rules), rules_parse_time) =
-  (* if there are no rules but just skipped rules, better to return an exn *)
-  (match (rules, skipped_rules) with
+let semgrep_with_rules config ((rules, invalid_rules), rules_parse_time) =
+  (* Return an exception
+     - always, if there are no rules but just invalid rules
+     - when users want to fail fast, if there are valid and invalid rules *)
+  (* TODO right now there is no option to not fail fast *)
+  (match (rules, invalid_rules) with
   | [], [] -> ()
   | [], err :: _ -> raise (Rule.InvalidRule err)
+  | _, err :: _ (* TODO fail fast when only when strict? *) ->
+      raise (Rule.InvalidRule err)
   | _ -> ());
```

Comment:
I don't think you need this case, but instead of `| [], [] -> ()` you could just write `| _, [] -> ()`.